pkg/tfgen/internal/unrec/testdata/TestSimplify.golden

```diff
@@ -0,0 +1,117 @@
+    },
+    "python": {
+      "compatibility": "tfbridge20",
+      "pyproject": {},
+      "readme": "\u003e This provider is a derived work of the [Terraform Provider](https://github.com/terraform-providers/terraform-provider-myprov)\n\u003e distributed under [MPL 2.0](https://www.mozilla.org/en-US/MPL/2.0/). If you encounter a bug or missing feature,\n\u003e first check the [`pulumi-myprov` repo](/issues); however, if that doesn't turn up anything,\n\u003e please consult the source [`terraform-provider-myprov` repo](https://github.com/terraform-providers/terraform-provider-myprov/issues)."
+    }
+  },
+  "config": {},
+  "types": {
+    "myprov:index/WebAclStatement:WebAclStatement": {
+      "properties": {
+        "andStatement": {
+          "$ref": "#/types/myprov:index/WebAclStatementAndStatement:WebAclStatementAndStatement"
+        },
+        "rateBasedStatement": {
+          "$ref": "#/types/myprov:index/WebAclStatementRateBasedStatement:WebAclStatementRateBasedStatement"
+        }
+      },
+      "type": "object"
+    },
+    "myprov:index/WebAclStatementAndStatement:WebAclStatementAndStatement": {
+      "properties": {
+        "statements": {
+          "type": "array",
+          "items": {
+            "$ref": "#/types/myprov:index/WebAclStatementAndStatementStatement:WebAclStatementAndStatementStatement"
+          }
+        }
+      },
+      "type": "object",
+      "required": [
+        "statements"
+      ]
+    },
+    "myprov:index/WebAclStatementAndStatementStatement:WebAclStatementAndStatementStatement": {
+      "properties": {
+        "andStatement": {
+          "$ref": "#/types/myprov:index/WebAclStatementAndStatement:WebAclStatementAndStatement"
+        },
+        "xssMatchStatement": {
+          "$ref": "#/types/myprov:index/WebAclStatementAndStatementStatementXssMatchStatement:WebAclStatementAndStatementStatementXssMatchStatement"
+        }
+      },
+      "type": "object"
+    },
+    "myprov:index/WebAclStatementAndStatementStatementXssMatchStatement:WebAclStatementAndStatementStatementXssMatchStatement": {
+      "properties": {
+        "fieldToMatch": {
+          "$ref": "#/types/myprov:index/WebAclStatementAndStatementStatementXssMatchStatementFieldToMatch:WebAclStatementAndStatementStatementXssMatchStatementFieldToMatch"
+        }
+      },
+      "type": "object"
+    },
+    "myprov:index/WebAclStatementAndStatementStatementXssMatchStatementFieldToMatch:WebAclStatementAndStatementStatementXssMatchStatementFieldToMatch": {
+      "properties": {
+        "method": {
+          "$ref": "#/types/myprov:index/WebAclStatementAndStatement:WebAclStatementAndStatement"
```

Comment:
This isn't right, because it used to be an empty object. Need to correct handling of empty object types that trivially satisfy `A <= B`.